zerver/tests/test_auth_backends.py

```diff
@@ -544,6 +544,7 @@ def _mock_key(self: RateLimitedAuthenticationByUsername) -> str:
 
         def attempt_authentication(username: str, password: str) -> Optional[UserProfile]:
             request = HttpRequest()
+            request.session = self.client.session
```

Comment:
In this particular place I don't think setting the session to the client session makes much sense? Could we just generate a fresh session here?